specs/version_format_specification.md

```diff
@@ -21,6 +21,8 @@ A version string must use the following character set:
 - Minus (`-`) separates the version and release parts.
 - Tilde (`~`) is a prefix that always compares lower.
 - Caret (`^`) is a prefix that always compares higher.
+- Plus (`+`) is permitted but ignored.
+- Underscore (`_`) is permitted but ignored, and it is suggested to avoid it.
```

Comment:
I would suggest the following instead:
```
The version string is a sequence of zero or more characters.

The following characters have special meaning:
- ASCII digits (`0-9`) form numerical components.
- ASCII letters (`a-z`, `A-Z`) form alphabetical components.
- Dot (`.`) separates parts of a component.
- Minus (`-`) separates major parts of the version string.
- Tilde (`~`) starts a suffix that always sorts lower.
- Caret (`^`) starts a suffix that always sorts higher.

Other characters are treated as separators.
This includes plus (`+`) and underscore (`_`) and other printable or non-printable characters.
The underscore MAY be used.
The plus SHOULD NOT be used, to avoid confusion with SEMVER which attaches a special meaning to it.
Other characters MUST NOT be used in a version string.

Note that in some contexts (for example <link to DDI spec>, DEB package file names), the underscore is used as a separator and cannot be used freely in the version string.
```

This changes the text to explicitly say that other characters are disallowed.
Fedora allows `[a-zA-Z0-9.~^_-]` (*), Debian the same except for `_` and `^` (
https://www.debian.org/doc/debian-policy/ch-controlfields.html#source, https://www.debian.org/doc/debian-policy/ch-controlfields.html#version), semver also a subset. So I think we can do this narrowing, because all package name-versions in distributions satisfy this requirement.

(*) The docs in Fedora could certainly be improved.
https://docs.fedoraproject.org/en-US/packaging-guidelines/Naming/#_common_character_set_for_package_naming describes the Name, but https://docs.fedoraproject.org/en-US/packaging-guidelines/Versioning/ isn't so nice as to list what characters are allowed and is a huge wall of text.

Reply:
looks great, updated and added Co-authored-by